examples/field_formats_example/public/plugin.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { KBN_FIELD_TYPES } from '@kbn/field-types';
+import {
+  AppMountParameters,
+  AppNavLinkStatus,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+} from '../../../src/core/public';
+import { DeveloperExamplesSetup } from '../../developer_examples/public';
+import { App } from './app';
+import { FieldFormatsSetup, FieldFormatsStart } from '../../../src/plugins/field_formats/public';
+import { registerExampleFormat } from './examples/2_creating_custom_formatter';
+import {
+  IndexPatternFieldEditorStart,
+  IndexPatternFieldEditorSetup,
+} from '../../../src/plugins/index_pattern_field_editor/public';
+import { DataPublicPluginStart } from '../../../src/plugins/data/public';
+import { registerExampleFormatEditor } from './examples/3_creating_custom_format_editor';
+import img from './formats.png';
+
+interface SetupDeps {
+  developerExamples: DeveloperExamplesSetup;
+  fieldFormats: FieldFormatsSetup;
+  indexPatternFieldEditor: IndexPatternFieldEditorSetup;
+}
+
+interface StartDeps {
+  fieldFormats: FieldFormatsStart;
+  indexPatternFieldEditor: IndexPatternFieldEditorStart;
+  data: DataPublicPluginStart;
+}
+
+export class FieldFormatsExamplePlugin implements Plugin<void, void, SetupDeps, StartDeps> {
+  public setup(core: CoreSetup<StartDeps>, deps: SetupDeps) {
+    registerExampleFormat(deps.fieldFormats);
+    registerExampleFormatEditor(deps.indexPatternFieldEditor);
+
+    // just for demonstration purposes:
+    // opens a field editor using default index pattern and first number field
+    const openIndexPatternNumberFieldEditor = async () => {
+      const [, plugins] = await core.getStartServices();
+      const indexPattern = await plugins.data.indexPatterns.getDefault();
+      if (!indexPattern) {
+        alert('Creating at least one index pattern to continue with this example');
```

Comment:
could we render these messages to the dom instead of displaying alerts?

Reply:
Hmm, I am actually happy with this usage 😃  I think this is a perfectly valid use case for an alert. 

I'd agree, that we shouldn't leave such alerts in user-facing code, but in artificial developer examples, especially in a helper-non-example code, I think it is fine.

I basically just want to pass a function to open the editor, to use as a shortcut to preview field formats inside the index pattern editor. I don't want to bloat example code with even more UI code not focused on the stuff we are demonstrating.